runelite-client/src/main/java/net/runelite/client/plugins/clanchat/ClanChatPlugin.java

```diff
@@ -70,22 +97,132 @@ public void updateClanChatTitle()
+			currentSetFilter = config.getTotalLevelFilter();
+		}
+
 		if (client.getGameState() != GameState.LOADING && client.getGameState() != GameState.LOGGED_IN)
 		{
 			return;
 		}
 
 		if (setMessage.getType() == ChatMessageType.CLANCHAT && client.getClanChatCount() > 0)
 		{
-			insertClanRankIcon(setMessage);
+			//Look through hash for acct if already to be filtered
+			if (isPlayerFiltered(setMessage.getName()))
+			{
+				log.debug("player: " +setMessage.getName() + " is already filtered");
+				filterMessage(setMessage);
+			}
+			//lookup rank and hiscore lookup if TL filter is active and account is not ranked in cc
+			else
+			{
+				ranked = insertClanRankIcon(setMessage);	//Check for ranked
+				if (config.getTotalLevelFilter() != 0 && !ranked)
+				{
+					executor.submit(() -> lookupPlayer(setMessage));
+					return;
+				}
+			}
+			client.refreshChat();
+		}
+	}
+
+	/**
+	 * Looks up player on hiscores and alters the message if it is to be filtered.
+	 * @param message account name pulled from message and altered if neede
+	 */
+	private void lookupPlayer(final SetMessage message)
+	{
+		String account = message.getMessageNode().getName();
+
+		account = sanitize(account);
+
+		HiscoreSkill skill;
+		skill =  HiscoreSkill.valueOf(net.runelite.api.Skill.OVERALL.getName().toUpperCase());
+
+		if (Strings.isNullOrEmpty(account))
+		{
+			return;
+		}
+		try
+		{
+			SingleHiscoreSkillResult result = hiscoreClient.lookup(account, skill);
+			Skill hiscoreSkill = result.getSkill();
+
+			if (config.getTotalLevelFilter() > hiscoreSkill.getLevel())
+			{
+				//Add account to "blacklist" hashSet
+				filteredAccounts.add(message.getMessageNode().getName());
+				filterMessage(message);
+			}
+		}
+		catch (IOException ex)
+		{
+			log.warn("Error fetching Hiscore data " + ex.getMessage());
+		}
+		//If player is not on hiscores this will throw a nullpointer exception, therefore we filter the message
+		catch (NullPointerException n)
+		{
+			log.debug("Null pointer");
+			filterMessage(message);
+		}
+		client.refreshChat();
+	}
+
+	/**
+	 * Alters the message to state that it has been filtered. Looking for way to just delete the message
+	 * instead of altering the contents.
+	 * @param message message to be altered
+	 */
+	private void filterMessage(SetMessage message)
+	{
+		message.getMessageNode().setName("FILTERED");
+		message.getMessageNode().setValue("");
+	}
+
+	/**
+	 * Looks through hashSet to see if the account has already been filtered.
+	 * @param name name of account to search
+	 * @return true if account has previously been filtered.
+	 */
+	private boolean isPlayerFiltered(String name)
+	{
+		log.debug("Looking for account: " + name);
+		for (String filtered: filteredAccounts)
+		{
+			log.debug(filtered);
+			if (name.contains(filtered))
+			{
+				return true;
+			}
 		}
+		return false;
 	}
 
 
-	private void insertClanRankIcon(final SetMessage message)
+	private String sanitize(String lookup)
```

Comment:
I am not sure what you try to do here, but if `message.getMessageNode().getName()` doesn't return clean name then should do changes there. 

You should write unit tests for this as well